### DISCUSSION TITLE
python-snap7 on python:3.9-alpine3.13 AND on ubuntu in both cases it gives RuntimeERROR

### DISCUSSION THREAD

**Discussion by Rahul14189** [gijzelaerr/python-snap7#462] [Q&A] [answered]
I have tried installing python-snap7 on python:3.9-alpine3.13, it gets installed, as i am able to import snap7, however, when i run the command "client = snap7.client.Client()", it throws a runtime error, stating
![image](https://github.com/gijzelaerr/python-snap7/assets/99563180/3fbc19a5-bf3a-4dca-8c67-09fe7c0c926b)

I am using Docker, so i tried running RUN ldconfig, after the installation, but alpine does not support this, i tried setting up the environment variable too, but the error remains the same,

I tried with ubuntu, again it gets installed, as i am able to import snap7, however, when i run the command "client = snap7.client.Client()", it throws a runtime error, same as above,  i tried same thing as above related to ldconfig and environment variable but the error remains the same,

For snap7 installation i am using pip install python-snap7 through Docker, 
However, i have tested it on Windows, it works fine but gives error in case of alpine and ubuntu

**Comment by nikteliy** [comment, marked as answer]
> alpine

We don't support musl-based platforms. If you want to use Alpine, you need to compile [snap7](https://snap7.sourceforge.net/) on [your own](https://python-snap7.readthedocs.io/en/latest/installation.html#manual-installation-not-recommended).

>  ubuntu

I've checked it on the `ubuntu:latest` image. It works

![ubuntu_latest](https://github.com/gijzelaerr/python-snap7/assets/52915342/ee9d0e19-8a3e-4513-bd59-25079118ab9a)

**Reply by Rahul14189** [reply]
Thanks a lot, it worked well at my end too
